### ISSUE TITLE
Install failure

### ISSUE THREAD

**Issue by rafaelgou** [Glavin001/atom-beautify#573]
Having an issue on install, any clues?

~~~
apm install atom-beautify
Installing atom-beautify to /root/.atom/packages ✗

> spawn-sync@1.0.13 postinstall /tmp/apm-install-dir-115825-14610-12vgb1i/node_modules/atom-beautify/node_modules/cross-spawn/node_modules/spawn-sync
> node postinstall

module.js:340
    throw err;
          ^
Error: Cannot find module '/tmp/apm-install-dir-115825-14610-12vgb1i/node_modules/atom-beautify/node_modules/cross-spawn/node_modules/spawn-sync/postinstall'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:935:3
npm ERR! Linux 3.19.0-28-generic
npm ERR! argv "/usr/share/atom/resources/app/apm/bin/node" "/usr/share/atom/resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/root/.atom/.apm/.apmrc" "--userconfig" "/root/.atom/.apmrc" "install" "/tmp/d-115825-14610-1hd7jq8/package.tgz" "--target=0.30.6" "--arch=x64"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! spawn-sync@1.0.13 postinstall: `node postinstall`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the spawn-sync@1.0.13 postinstall script 'node postinstall'.
npm ERR! This is most likely a problem with the spawn-sync package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node postinstall
npm ERR! You can get their info via:
npm ERR!     npm owner ls spawn-sync
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /tmp/apm-install-dir-115825-14610-12vgb1i/npm-debug.log
~~~

**Comment by Glavin001** [comment]
This is a duplicate of many others.

I'd recommend taking a read through them: https://github.com/Glavin001/atom-beautify/search?q=spawn-sync&type=Issues&utf8=%E2%9C%93

**Comment by rafaelgou** [comment]
Thanks, @Glavin001 !

**Comment by Glavin001** [comment]
Also maybe take a look at https://github.com/ForbesLindesay/spawn-sync/issues/28#issuecomment-138992018

This is a problem with [`spawn-sync`](https://github.com/ForbesLindesay/spawn-sync) and [`apm` (Atom's package manager)](https://github.com/atom/apm). I'd recommend searching for an issue there and joining their discussions. Hope it works out! Feel free to post your working solutions here.

**Comment by rafaelgou** [comment]
Just for the records: the problem occurs when installing as superuser (root).

Works fine when doing as normal user. It seems to be a `npm` security feature that downgrades permissions.

https://github.com/ForbesLindesay/spawn-sync/issues/11#issuecomment-102750901

A simple workaround is to run as simpleuser and copy `.atom/packages/atom-beautify` to superuser's atom's folder.

**Comment by scamden** [comment]
this also causes problems with shrinkwrap because try-thread-sleep ends up in both dev and regular deps

**Comment by Glavin001** [comment]
I just published a patch in release v0.29.4 that makes `cross-spawn` dependency optional. 
The [`cross-spawn`](https://github.com/IndigoUnited/node-cross-spawn) package depends on [`spawn-sync`](https://github.com/ForbesLindesay/spawn-sync) and has been causing problems when installing Atom-Beautify for many users. By making this dependency optional, users will be able to easily install Atom-Beautify and, in the case that [`cross-spawn`](https://github.com/IndigoUnited/node-cross-spawn) fails to install, will fallback to the builtin [`spawn`](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options) method.
